providers/vault/vault.go

```diff
@@ -34,6 +34,9 @@ type Config struct {
 
 	// Internal HTTP client timeout
 	Timeout time.Duration
+
+	// WithMeta states whether the secret should be returned with its metadata.
```

Comment:
Please explicitly document here, the big change in the response structure when this flag is on/off with a couple examples (`eg: data.YOUR_KEY vs. data.KEY and data vs. metadata)

Reply:
Have updated the docs.